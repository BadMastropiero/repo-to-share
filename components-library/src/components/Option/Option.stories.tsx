import {ComponentMeta, ComponentStory} from '@storybook/react';
import {useState} from 'react';

import Option from './Option';

export default {
  title: 'Components/Option',
  component: Option,
} as ComponentMeta<typeof Option>;

export const Base: ComponentStory<typeof Option> = args => {
  const {primary, disabled, text, noCheckbox} = args;
  const [selected, setSelected] = useState<boolean>(false);

  return (
    <Option
      primary={primary}
      disabled={disabled}
      selected={selected}
      onClick={() => {
        setSelected(!selected);
      }}
      text={text}
      noCheckbox={noCheckbox}
    />
  );
};

Base.args = {
  primary: true,
  disabled: false,
  text: 'Option1',
};

Base.parameters = {
  design: {
    type: 'figma',
    url: 'https://www.figma.com/file/epHXucNHjke2rES69iOsaS/TetraImpacts?node-id=2%3A1511',
  },
};

export const NoCheckbox: ComponentStory<typeof Option> = args => {
  const {primary} = args;
  const [selected, setSelected] = useState<boolean>(false);

  return (
    <Option
      primary={primary}
      selected={selected}
      onClick={() => {
        setSelected(!selected);
      }}
      text="No checkbox in here"
      noCheckbox
    />
  );
};

NoCheckbox.args = {
  primary: true,
  disabled: false,
};

NoCheckbox.parameters = {
  design: {
    type: 'figma',
    url: 'https://www.figma.com/file/epHXucNHjke2rES69iOsaS/TetraImpacts?node-id=2%3A1511',
  },
};
