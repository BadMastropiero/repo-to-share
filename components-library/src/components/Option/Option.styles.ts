import styled, {css} from 'styled-components';

const OptionContainer = styled.div<{disabled?: boolean; primary?: boolean; noCheckbox?: boolean}>(
  ({
    theme: {
      colors: {primary: primaryColor, primary50, primary600, primaryBg},
      transition,
    },
    primary: isPrimary,
    noCheckbox,
    disabled,
  }) => css`
    position: relative;
    left: 0;
    top: 0;
    fill: none;
    display: flex;
    flex-direction: row;
    justify-content: ${noCheckbox ? 'flex-end' : 'space-between'};
    align-content: space-around;
    align-items: center;
    align-self: center;
    padding: 5px 10px 5px 5px;
    gap: 5px;
    height: 20px;
    background-color: ${isPrimary ? primaryColor : primaryBg};
    border-radius: 10px;
    /* cursor: ${disabled ? '' : 'pointer'}; */
    transition: ${transition};

    &:hover {
      fill: none;
      background-color: ${disabled || (isPrimary && primary600) || primary50};
    }
  `
);

export default OptionContainer;
