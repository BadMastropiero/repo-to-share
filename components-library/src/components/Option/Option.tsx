import React from 'react';
import {Typography} from '../Typography';
import {Checkbox} from '../Checkbox';

// Styles
import OptionContainer from './Option.styles';

// Interfaces
import {OptionProps} from './Option.interface';

const Option: React.FC<OptionProps> = ({
  selected,
  text,
  onClick,
  primary = false,
  disabled = false,
  noCheckbox,
}) => (
  <OptionContainer
    role="option"
    primary={primary}
    disabled={disabled}
    onClick={disabled ? undefined : onClick}
    noCheckbox={noCheckbox}
  >
    {noCheckbox || <Checkbox checked={selected} primary={primary} />}
    <Typography
      colorPreset="disable" // seems that is always same font color
      fontSize="12px"
      fontStyle="normal"
      fontWeight={selected ? '800' : '500'}
      lineHeight="14px"
      preset="content"
      textTransform="uppercase"
      textAlign="right"
      variant="p"
    >
      {text}
    </Typography>
  </OptionContainer>
);

export default Option;
