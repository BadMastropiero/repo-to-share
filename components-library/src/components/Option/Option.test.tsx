import {render, screen} from '@testing-library/react';
import {composeStories} from '@storybook/testing-react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import * as stories from './Option.stories';

// Every component that is returned maps 1:1 with the stories, but they already contain all decorators from story level, meta level and global level!
const {Base, NoCheckbox} = composeStories(stories);

describe('Option tests', () => {
  test('should render', () => {
    render(<Base />);
    expect(screen.getByRole('option')).toBeInTheDocument();
  });

  test('should match snapshot', () => {
    const tree = render(<Base />);
    expect(tree).toMatchSnapshot();
  });

  test('should toggle', () => {
    render(<Base />);

    const option = screen.getByRole('checkbox');

    // Execute the click event of the option
    userEvent.click(option);
    expect(option).toBeChecked();
    // expect(screen.queryByTestId("checkmark")).toBeInTheDocument();

    // Execute the click event again
    userEvent.click(option);
    expect(option).not.toBeChecked();
    // expect(screen.queryByTestId("checkmark")).not.toBeInTheDocument();
  });
});

describe('Option (nocheckbox) tests', () => {
  test('should render', () => {
    render(<NoCheckbox />);
    expect(screen.getByRole('option')).toBeInTheDocument();
  });

  test('should match snapshot', () => {
    const tree = render(<NoCheckbox />);
    expect(tree).toMatchSnapshot();
  });

  test('checkbox should not exist', () => {
    const {queryByTestId} = render(<NoCheckbox />);
    expect(queryByTestId('checkbox')).toBeNull();
  });
});
