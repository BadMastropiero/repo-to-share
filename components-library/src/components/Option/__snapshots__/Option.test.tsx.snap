// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Option (nocheckbox) tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gKPRtg hldrop"
        role="option"
      >
        <p
          class="sc-bcXHqe sc-gswNZR edCjtd"
          font-size="12px"
          font-style="normal"
          font-weight="500"
        >
          No checkbox in here
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gKPRtg hldrop"
      role="option"
    >
      <p
        class="sc-bcXHqe sc-gswNZR edCjtd"
        font-size="12px"
        font-style="normal"
        font-weight="500"
      >
        No checkbox in here
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Option tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gKPRtg hVXxrD"
        role="option"
      >
        <div
          class="sc-hLBbgP bOzfvb"
        >
          <input
            class="sc-jSUZER jDSytf"
            height="20px"
            type="checkbox"
            width="20px"
          />
        </div>
        <p
          class="sc-bcXHqe sc-gswNZR edCjtd"
          font-size="12px"
          font-style="normal"
          font-weight="500"
        >
          Option1
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gKPRtg hVXxrD"
      role="option"
    >
      <div
        class="sc-hLBbgP bOzfvb"
      >
        <input
          class="sc-jSUZER jDSytf"
          height="20px"
          type="checkbox"
          width="20px"
        />
      </div>
      <p
        class="sc-bcXHqe sc-gswNZR edCjtd"
        font-size="12px"
        font-style="normal"
        font-weight="500"
      >
        Option1
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
